WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift

```diff
@@ -0,0 +1,274 @@
+import UIKit
+import WordPressAuthenticator
+import SafariServices
+
+
+/// UI containing modals preented in the Card Present Payments flows.
+final class CardPresentPaymentsModalViewController: UIViewController {
+    /// The view model providing configuration for this view controller
+    /// and support for user actions
+    private var viewModel: CardPresentPaymentsModalViewModel
+
+    @IBOutlet private weak var topTitleLabel: UILabel!
+    @IBOutlet private weak var topSubtitleLabel: UILabel!
+    @IBOutlet private weak var bottomTitleLabel: UILabel!
+    @IBOutlet private weak var bottomSubtitleLabel: UILabel!
+
+    @IBOutlet private weak var primaryButton: NUXButton!
+    @IBOutlet private weak var secondaryButton: NUXButton!
+    @IBOutlet private weak var imageView: UIImageView!
+    @IBOutlet private weak var extraInfoButton: UIButton!
+
+    @IBOutlet private weak var actionButtonsView: UIView!
+    @IBOutlet private weak var bottomLabels: UIStackView!
+
+
+    init(viewModel: CardPresentPaymentsModalViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: Self.nibName, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        styleContent()
+        populateContent()
+    }
+
+    func setViewModel(_ newViewModel: CardPresentPaymentsModalViewModel) {
+        self.viewModel = newViewModel
+
+        populateContent()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
```

Comment:
Why is this needed? And viewWillTransition also?

Reply:
My bad, I will remove them. The reason I added them is because we will need this to support dynamic type and rotation properly (#4064)

* when rotating the device, we'll need to change the main stack view's axis from vertical to horizontal
* for certain font sizes we might want to tweak certain properties in the stack view.

We can do those things overriding this method here.